freqtrade/edge/__init__.py

```diff
@@ -150,10 +160,24 @@ def calculate(self) -> bool:
 
         return True
 
-    def stake_amount(self, pair: str) -> float:
+    def stake_amount(self, pair: str, free_capital: float, total_capital: float) -> float:
+        if pair not in self._cached_pairs:
+            logger.warning("cannot find %s in calculated pairs, "
+                           "stake_amount of strategy is used instead.", pair)
+            return self.config['stake_amount']
+
         stoploss = self._cached_pairs[pair].stoploss
-        allowed_capital_at_risk = round(self._total_capital * self._allowed_risk, 5)
-        position_size = abs(round((allowed_capital_at_risk / stoploss), 5))
+        available_capital = total_capital * self._capital_percentage
+        allowed_capital_at_risk = round(available_capital * self._allowed_risk, 15)
+        max_position_size = abs(round((allowed_capital_at_risk / stoploss), 15))
```

Comment:
why are you rounding to 15 in the middle of the calculation? 
wouldn't rounding for output (to fewer decimals) - and to 15 for the return value make more sense?

Reply:
the answer is yes :)